samcli/commands/check/resources/TemplateResource.py

```diff
@@ -0,0 +1,31 @@
+"""
+Super class for all resources
+"""
+from typing import Any
+from samcli.lib.providers.provider import Function
+
+
+class TemplateResource:
+    _resource_object: Function
+    _resource_type: str
+
+    def __init__(self, resource_object: Any, resource_type: str):
+        """
+        Args:
+            resource_object (Function): The resource object form the template file
+            resource_type (str): The resource type
+        """
+        self._resource_object = resource_object
+        self._resource_type = resource_type
+
+    @property
+    def resource_object(self) -> Function:
```

Comment:
The return type should match the type defined for `self._resource_object` (which is `Any`). Did `mypy` pass?

Reply:
Odd. I ran mypy but never got an error about that. Still, it needs to be changed

Reply:
oh, probably `self._resource_object` is annotated as `Function` on L9. 
But the input `reousrce_object` in `__init__` is `Any`